### ISSUE TITLE
Sampling step?

### ISSUE THREAD

**Issue by anshulsamar** [google-research/electra#87]
Thanks for your documentation and transparency here and nice to see more environmentally friendly models.

Quick question, in `sample_from_softmax(logits, disallow=None)`, you return:

`tf.one_hot(tf.argmax(tf.nn.softmax(logits + gumbel_noise), -1, output_type=tf.int32), logits.shape[-1])`

The original function is called by passing in `logits/temperature` for some temperature parameter.

Wondering why tf.softmax or temperature are needed here if the result will be passed through tf.argmax anyways? Perhaps this is a holdover from another experiment?

Thanks!

**Comment by Acejoy** [comment]
Hey, did you find the answer to this?

**Comment by anshulsamar** [comment]
Hi, I didn't, sorry
